packages/explorer/src/helpers/postMessageRelayHelpers.ts

```diff
@@ -195,28 +205,99 @@ export function executeOperation({
     }),
   })
     .then(async (response) => {
-      const json = await response.json();
-
       const responseHeaders: Record<string, string> = {};
       response.headers.forEach((value, key) => {
         responseHeaders[key] = value;
       });
 
-      sendPostMessageToEmbed({
-        message: {
-          // Include the same operation ID in the response message's name
-          // so the Explorer knows which operation it's associated with
-          name: EXPLORER_QUERY_MUTATION_RESPONSE,
-          operationId,
-          response: {
-            ...json,
-            status: response.status,
-            headers: responseHeaders,
+      const contentType = response.headers?.get('content-type');
+      if (contentType !== null && /^multipart\/mixed/.test(contentType)) {
+        const observable = new Observable<MultipartResponse>(
+          (observer: Observer<MultipartResponse>) =>
+            readMultipartWebStream(response, contentType, observer)
+        );
+
+        observable.subscribe({
+          next(data) {
+            sendPostMessageToEmbed({
+              message: {
+                // Include the same operation ID in the response message's name
+                // so the Explorer knows which operation it's associated with
+                name: EXPLORER_QUERY_MUTATION_RESPONSE,
+                operationId,
+                response: {
+                  data: data.data.data,
+                  errors: data.data.errors,
+                  extensions: data.data.extensions,
+                  path: data.data.path,
+                  status: response.status,
+                  headers: responseHeaders,
+                  hasNext: true,
+                  size: data.size,
+                },
+              },
+              embeddedIFrameElement,
+              embedUrl,
+            });
           },
-        },
-        embeddedIFrameElement,
-        embedUrl,
-      });
+          error(err) {
+            sendPostMessageToEmbed({
+              message: {
+                // Include the same operation ID in the response message's name
+                // so the Explorer knows which operation it's associated with
+                name: EXPLORER_QUERY_MUTATION_RESPONSE,
+                operationId,
+                response: {
+                  data: null,
+                  error: { message: err },
```

Comment:
is err a string here? I would have assumed it would likely be an Error object?

Reply:
Fixed 👍